src/basic/macro.h

```diff
@@ -429,4 +429,10 @@ static inline unsigned long ALIGN_POWER2(unsigned long u) {
                         func(*p);                               \
         }
 
+/* A macro to be used in enum-based bit mask definitions. Why bother with this? This makes sure the shift left doesn't
+ * result in sign extension or overruns because sizeof(1) is 4 on our architectures. Note that in C the type of a enum
+ * item value doesn't affect the size chose for the enum, hence even if we use the ULL suffix here this doesn't mean
+ * the flags field grows to 64bit all the time. It just makes sure the shift left is an unsigned one, with lots of bits */
+#define ENUM_BIT(x) (1ULL << (x))
```

Comment:
Why is this named `ENUM_BIT()` rather than simply `BIT()`?  The chosen name would suggest that it does something specific to enum types, but all it really does is to set a bit. Not that consistency maters much across projects, but the Linux kernel has a `BIT()` macro with the same definition.

Reply:
it *is* specific to enum types, as doing the ULL suffix is not a good idea otherwise, as in other contexts it will upgrade everything to an unsigned long long. Only in enum context this is a safe choice, as only then the type of the field actually does not matter, as the enum type will be derived from the values, not the types...